src/io/iter_prefetcher.h

```diff
@@ -79,17 +79,22 @@ class PrefetcherIter : public IIterator<DataBatch> {
           (*dptr)->data.resize(batch.data.size());
           (*dptr)->index.resize(batch.batch_size);
           for (size_t i = 0; i < batch.data.size(); ++i) {
+            auto dtype = param_.dtype != mshadow::default_type_flag
```

Comment:
But user can also specify it to be float32.
How about change default of param_.dtype to -1?